src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs

```diff
@@ -1203,24 +1186,29 @@ public RunningBackup OnGoingBackup(long taskId)
             };
         }
 
-        public IReadOnlyDictionary<string, long> GetLastProcessedDocumentTombstonesPerCollection()
+        public Dictionary<string, long> GetLastProcessedDocumentTombstonesPerCollection()
         {
-            var results = new Dictionary<string,long>();
+            var results = new Dictionary<string, long>();
 
+            var minLastEtag = long.MaxValue;
             foreach (var periodicBackup in _periodicBackups.Values)
             {
-                foreach (var etagItem in periodicBackup.BackupStatus.LastTombstoneEtagsByCollection)
+                if (periodicBackup.BackupStatus?.LastEtag != null &&
+                    minLastEtag > periodicBackup.BackupStatus?.LastEtag)
                 {
-                    if (results.TryGetValue(etagItem.Key, out long existingEtag))
-                    {
-                        //since we are collecting max etags for tombstones cleaner,
-                        //we need to get the minimal etag processed in all backups
-                        results[etagItem.Key] = Math.Min(existingEtag, etagItem.Value);
-                    }
-                    else
-                    {
-                        results[etagItem.Key] = etagItem.Value;
-                    }
+                    minLastEtag = periodicBackup.BackupStatus.LastEtag.Value;
+                }
+            }
+
+            if(minLastEtag == long.MaxValue)
+                minLastEtag = 0;
+
+            using (_database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                foreach (var collection in _database.DocumentsStorage.GetCollections(context).ToList())
```

Comment:
Why the `ToList` here?

Reply:
No reason - forgot to remove it. Will remove with next PR